### ISSUE TITLE
backup cannot be synchronized normally

### ISSUE THREAD

**Issue by ShyunnY** [vmware-tanzu/velero#8040]
**What steps did you take and what happened:**
I am a novice using velero. I followed the steps below:
+ I created a backup to back up resources: `./velero backup create envoy-backup --include-namespaces envoy-gateway-system`
Then I used the `velero` command line tool to view the results(which looks great! I also checked that the corresponding data exists in S3)
```shell
./velero backup get
NAME STATUS ERRORS WARNINGS CREATED EXPIRES STORAGE LOCATION SELECTOR
envoy-backup Completed 0 0 2024-07-23 19:20:21 +0800 CST 29d default <none>
```
+ Now I want to restore the data, I do the following (this looks bad, the backup is gone for some reason!):
```shell
./velero restore create envoy-restore --from-backup envoy-backup
An error occurred: backups.velero.io "envoy-backup" not found
```

+ I confirmed again that the corresponding Backup data was stored in my S3, and then I tried to find the reason from the log, but I couldn’t:
```shell
time="2024-07-23T11:25:08Z" level=info msg="plugin process exited" backup-storage-location=velero/default cmd=/plugins/velero-plugin-for-aws controller=backup-storage-location id=1966 logSource="pkg/plugin/clientmgmt/process/logrus_adapter.go:80" plugin=/plugins/velero-plugin-for-aws
time="2024-07-23T11:25:17Z" level=info msg="Validating BackupStorageLocation" backup-storage-location=velero/default controller=backup-storage-location logSource="pkg/controller/backup_storage_location_controller.go:141"
time="2024-07-23T11:25:17Z" level=info msg="BackupStorageLocations is valid, marking as available" backup-storage-location=velero/default controller=backup-storage-location logSource="pkg/controller/backup_storage_location_controller.go:126"
time="2024-07-23T11:25:17Z" level=info msg="plugin process exited" backup-storage-location=velero/default cmd=/plugins/velero-plugin-for-aws controller=backup-storage-location id=1979 logSource="pkg/plugin/clientmgmt/process/logrus_adapter.go:80" plugin=/plugins/velero-plugin-for-aws
time="2024-07-23T11:25:27Z" level=info msg="Validating BackupStorageLocation" backup-storage-location=velero/default controller=backup-storage-location logSource="pkg/controller/backup_storage_location_controller.go:141"
time="2024-07-23T11:25:27Z" level=info msg="BackupStorageLocations is valid, marking as available" backup-storage-location=velero/default controller=backup-storage-location logSource="pkg/controller/backup_storage_location_controller.go:126"
time="2024-07-23T11:25:27Z" level=info msg="plugin process exited" backup-storage-location=velero/default cmd=/plugins/velero-plugin-for-aws controller=backup-storage-location id=1992 logSource="pkg/plugin/clientmgmt/process/logrus_adapter.go:80" plugin=/plugins/velero-plugin-for-aws
time="2024-07-23T11:25:37Z" level=info msg="Validating BackupStorageLocation" backup-storage-location=velero/default controller=backup-storage-location logSource="pkg/controller/backup_storage_location_controller.go:141"
time="2024-07-23T11:25:37Z" level=info msg="BackupStorageLocations is valid, marking as available" backup-storage-location=velero/default controller=backup-storage-location logSource="pkg/controller/backup_storage_location_controller.go:126"
time="2024-07-23T11:25:37Z" level=info msg="plugin process exited" backupLocation=velero/default cmd=/plugins/velero-plugin-for-aws controller=backup-sync id=2005 logSource="pkg/plugin/clientmgmt/process/logrus_adapter.go:80" plugin=/plugins/velero-plugin-for-aws
time="2024-07-23T11:25:37Z" level=info msg="plugin process exited" backup-storage-location=velero/default cmd=/plugins/velero-plugin-for-aws controller=backup-storage-location id=2019 logSource="pkg/plugin/clientmgmt/process/logrus_adapter.go:80" plugin=/plugins/velero-plugin-for-aws
time="2024-07-23T11:25:47Z" level=info msg="Validating BackupStorageLocation" backup-storage-location=velero/default controller=backup-storage-location logSource="pkg/controller/backup_storage_location_controller.go:141"
time="2024-07-23T11:25:47Z" level=info msg="BackupStorageLocations is valid, marking as available" backup-storage-location=velero/default controller=backup-storage-location logSource="pkg/controller/backup_storage_location_controller.go:126"
time="2024-07-23T11:25:47Z" level=info msg="plugin process exited" backup-storage-location=velero/default cmd=/plugins/velero-plugin-for-aws controller=backup-storage-location id=2030 logSource="pkg/plugin/clientmgmt/process/logrus_adapter.go:80" plugin=/plugins/velero-plugin-for-aws
time="2024-07-23T11:25:57Z" level=info msg="Validating BackupStorageLocation" backup-storage-location=velero/default controller=backup-storage-location logSource="pkg/controller/backup_storage_location_controller.go:141"
time="2024-07-23T11:25:57Z" level=info msg="BackupStorageLocations is valid, marking as available" backup-storage-location=velero/default controller=backup-storage-location logSource="pkg/controller/backup_storage_location_controller.go:126"
time="2024-07-23T11:25:58Z" level=info msg="plugin process exited" backup-storage-location=velero/default cmd=/plugins/velero-plugin-for-aws controller=backup-storage-location id=2042 logSource="pkg/plugin/clientmgmt/process/logrus_adapter.go:80" plugin=/plugins/velero-plugin-for-aws
time="2024-07-23T11:26:07Z" level=info msg="Validating BackupStorageLocation" backup-storage-location=velero/default controller=backup-storage-location logSource="pkg/controller/backup_storage_location_controller.go:141"
time="2024-07-23T11:26:07Z" level=info msg="BackupStorageLocations is valid, marking as available" backup-storage-location=velero/default controller=backup-storage-location logSource="pkg/controller/backup_storage_location_controller.go:126"
time="2024-07-23T11:26:08Z" level=info msg="plugin process exited" backup-storage-location=velero/default cmd=/plugins/velero-plugin-for-aws controller=backup-storage-location id=2055 logSource="pkg/plugin/clientmgmt/process/logrus_adapter.go:80" plugin=/plugins/velero-plugin-for-aws
```

**What did you expect to happen:**

I don't know what caused my Backup to be lost, which makes me very upset. I saw in the documentation: Velero's Backup will be synchronized with S3, but I don't know why Velero doesn't do this.
I hope to be able to back up the data normally.

**Environment:**

- Velero version (`v1.14.0`): 
- Velero features (use `features: <NOT SET>`): 
- Kubernetes version (use `Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.11", GitCommit:"38d3c1f3d5306401bcf39a71bad3b5a5106033d7", GitTreeState:"clean", BuildDate:"2022-03-16T14:02:06Z", GoVersion:"go1.16.15", Compiler:"gc", Platform:"linux/amd64"}`):

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

**Comment by ShyunnY** [comment]
One more thing: I found that `Backup` was deleted in about 30s

**Comment by blackpiglet** [comment]
There are several possible reasons:
* Please check whether any GitOps is governing your k8s environment. It's possible the generated backup was deleted by the GitOps tools
* If you use Velero with an on-premise Object Storage service, please check whether the BackupStorageLocation's `s3ForcePathStyle` parameter is set to `true`.  https://velero.io/docs/v1.14/contributions/minio/#set-up-server

**Comment by ShyunnY** [comment]
> There are several possible reasons:
> 
> * Please check whether any GitOps is governing your k8s environment. It's possible the generated backup was deleted by the GitOps tools
> * If you use Velero with an on-premise Object Storage service, please check whether the BackupStorageLocation's `s3ForcePathStyle` parameter is set to `true`.  https://velero.io/docs/v1.14/contributions/minio/#set-up-server

Thanks for your reply!
Unfortunately, I have tried both solutions above and it seems that the problem cannot be solved.

I set up `minio` in my local cluster and tried the quick-start example. This is normal, and my `Backup` runs as expected.
I don't know what happened, but I'm eager to solve this problem.
> In addition, the s3 I use is Qiniu Cloud Kodo storage

**Comment by blackpiglet** [comment]
Let me sum up the information:
* If the object storage is MinIO, the backup works.
* If the object storage is Kodo, the backup was deleted 30 seconds after creation.
* Although the backup was deleted from the k8s cluster, the backup metadata can be found in the Kodo storage.

IMO, this is dedicated to Qiniu Kodo storage. 
Does the Kodo storage support using the path-style URL?
https://docs.aws.amazon.com/AmazonS3/latest/userguide/VirtualHosting.html

**Comment by ShyunnY** [comment]
@blackpiglet 

yeah, we can see from the documentation that it does support this feature:  [docs](https://developer.qiniu.com/kodo/4088/s3-access-domainname)

**Comment by ShyunnY** [comment]
Thanks :)

I solved this problem. After carefully analyzing the workflow of BackupController, I found that: Since BackupController has a cycle (seemingly called the synchronization cycle) to obtain the Backup in S3 storage, and since BackupController cannot obtain the corresponding resources from S3 within a cycle, the Backup CR is deleted during reconcile.

Based on this idea: I finally found that _it was because my S3 URL was not a complete PathStyle style_, so BackupController could not obtain the Backup resource as expected. (But I was confused, even if the S3 URL was wrong, the Backup could still be stored in S3)
